wp-cache-phase2.php

```diff
@@ -530,8 +569,10 @@ function get_current_url_supercache_dir( $post_id = 0 ) {
 	} else {
 		$uri = strtolower( $wp_cache_request_uri );
 	}
-	$uri = wpsc_deep_replace( array( '..', '\\', 'index.php', ), preg_replace( '/[ <>\'\"\r\n\t\(\)]/', '', preg_replace( "/(\?.*)?$/", '', $uri ) ) );
-	$dir = preg_replace( '/:.*$/', '',  $WPSC_HTTP_HOST ) . $uri; // To avoid XSS attacks
+	$uri = wpsc_deep_replace( array( '..', '\\', 'index.php', ), preg_replace( '/[ <>\'\"\r\n\t\(\)]/', '', preg_replace( "/(\?.*)?(#.*)?$/", '', $uri ) ) );
+	// Get hostname from wp options for wp-cron, wp-cli and similar requests.
+	$hostname = empty( $WPSC_HTTP_HOST ) ? (string) parse_url( get_option( 'home' ), PHP_URL_HOST ) : $WPSC_HTTP_HOST;
```

Comment:
This could be a problem for users running old versions of WordPress as get_option() won't always be defined. I tested WordPress 4.4 as get_current_url_supercache_dir() is called by wp_cache_serve_cache_file() and that obviously runs really early on. (I didn't test your patch directly, I Just added a function_exists( 'get_option' ) to get_current_url_supercache_dir() and it returned false.)

We probably need to check if get_option() exists and if it doesn't and if WPSC_HTTP_HOST is blank disable caching. What do you think?

Reply:
I just added checking if function `get_option` exists. So, it'll help because almost all cases are related (when WPSC_HTTP_HOST is empty) for cron request (via php-cli), wp-cli or other kind of php-cli scripts. I'm planning to make some improvements related to preload and for now it's enough (my original idea is to fix issues related to cron jobs).
It's very rare case that webserver doesn't pass $_SERVER variables. So, I think that it's acceptable for merging now. Anyway, if someone uses very old webserver, php and wordpress, then there are many other issue and he should update everything.

Reply:
Thanks! Unfortunately WP 4.4 is still maintained so it's going to be an issue for a long time but I agree it should hardly ever happen.

I'm excited to get this into the plugin. It's a really nice cleanup of some of the code!